vignettes/pkgnet-intro.Rmd

```diff
@@ -0,0 +1,249 @@
+
+```{r nodes}
+dim(reportObjWithCovr$PackageFunctionReporter$nodes)
+names(reportObjWithCovr$PackageFunctionReporter$nodes)
+```
+
+### Network Measures 
+Both the `PackageDependencyReporter` and the `PackageFunctionReporter` contain measure based on their network structure in a `network_measures` list. 
+
+```{r networkMeasures}
+reportObjWithCovr$PackageFunctionReporter$network_measures
+```
+
+### Network Object
+Both the `PackageDependencyReporter` and the `PackageFunctionReporter` are available as [igraph](http://igraph.org/r/) objects named `pkg_graph`
+
+```{r networkObj}
+reportObjWithCovr$PackageFunctionReporter$pkg_graph
+```
+
+***
+# A Deeper Look
+With the reports and objects produced by `pkgnet` by default, there is plenty to inform us on the inner workings of an R package.  However, we may want to know MORE! Since the [igraph](http://igraph.org/r/) objects are available, we can leverage those for further analysis.  Here are some examples.  
+
+## Coverage of Most Referenced Functions
+Let's visualize the relationship between centrality and coverage in a large package such as `lubridate`.
+
+<!--  Faked Since lubridate not guaranteed to be installed at time of report creation -->
+```{r fakeDetail1, eval=FALSE}
+# Run pkgnet
+library(pkgnet)
+reportObjWithCovr <- CreatePackageReport(pkg_name = "lubridate"
+                                         , pkg_path = "~/repos/lubridate")
+
+# Extract Nodes Table
+funcNodes <- reportObjWithCovr$PackageFunctionReporter$nodes
+
+# List Coverage For Most Referenced Functions
+mostRef <- funcNodes[order(numDescendants, decreasing = TRUE)][1:10]
+mostRef[,list(node, numDescendants, coverageRatio, totalLines)]
+
+```
+
+|node                   | numDescendants| coverageRatio| totalLines|
+|:----------------------|--------------:|-------------:|----------:|
+|dmicroseconds          |             39|     0.0000000|          1|
+|days_in_months_so_far  |             22|     0.0000000|         10|
+|check_period           |             15|     0.0000000|         12|
+|as.difftime            |             13|            NA|         NA|
+|fast_strptime          |             13|     0.8571429|          7|
+|comp_posix_date        |             12|     0.7692308|         13|
+|date_to_posix          |             12|     1.0000000|          2|
+|ceiling_date           |             10|     0.9666667|         30|
+|add_number_to_duration |              8|     1.0000000|          1|
+|check_interval         |              8|     0.0000000|         11|
+
+Inspecting results such as these can help an R package developer decide which function to cover with unit tests next.  In this case, the most referenced function, `dmicroseconds`, appears to be a simple one line function that may not be necessary to cover in unit testing.  However, the second most referenced function, `days_in_months_so_far`, might benefit from some unit test coverage.
```

Comment:
this is clear to me, but I think that's because I'm a part of this project and used to thinking this way.

I propose adding a paragraph somewhere in this doc (wherever you think fits) that looks like this:

> Your development time is limited...you can't write tests for everything today. `pkgnet` can help you decide which things are most worth your investment of time and effort. There are two strategies that you might take:

> 1) test the function which has the highest out-degree. While this might not technically help you in coverage statistics, you would effectively be testing code that gets called many many times when your package is used

> 2) test the function which is downstream of the most other functions. Getting this right means that you get at least _some_ coverage of all the other functions that it depends on

Reply:
That brings up another point...I think we need a dedicated section at the beginning describing how `pkgnet` models dependency graphs. Like `numDescendants` only makes sense down here if you know that the graph for functions is directed and that the relation modeled by an edge is "relies on" vs. "is used by"

Reply:
Good point. That is a glaring missing section.  Should probably be in the read me too.  

Also, I'll add that time savings bit somewhere.  Also sounds like it should be in the readme